src/append_only_data.rs

```diff
@@ -132,6 +163,7 @@ impl PubPermissionSet {
         self.manage_permissions = manage_perms.into();
     }
 
+    /// Returns `true` if `action` is allowed.
```

Comment:
Returns an `Option<bool>`, so it should be something like
```suggestion
    /// Returns `Some(true)` if `action` is allowed and `Some(false)` if it's not permitted.
    /// `None` would mean that default permissions should be applied.
```